### ISSUE TITLE
[Bug]: Cliend side routing doesn't work

### ISSUE THREAD

**Issue by abereghici** [remix-run/remix#1150]
### Which Remix packages are impacted?

- [X] `remix` (Remix core)
- [ ] `create-remix`
- [ ] `@remix-run/architect`
- [ ] `@remix-run/cloudflare-workers`
- [ ] `@remix-run/dev`
- [ ] `@remix-run/express`
- [ ] `@remix-run/netlify`
- [ ] `@remix-run/node`
- [ ] `@remix-run/react`
- [ ] `@remix-run/serve`
- [ ] `@remix-run/server-runtime`
- [ ] `@remix-run/vercel`

### What version of Remix are you using?

1.1.1

### What version of Node are you using? Minimum supported version is 14.

v14.15.4

### Steps to Reproduce

Hey. I just tried the latest version ( 1.1.1 ) and I noticed that the client side routing doesn't work anymore. Is it expected ? I mean, this link will reload the entire page. I just tested 1.0.6 version and there is working as expected.
import {Link} from 'remix'

export default function TestIndex() {
  return (
    <div>
      <h1>Test page</h1>
      <Link to="/">Go to home page</Link>
    </div>
  )
}

### Expected Behavior

I'm expecting to have the client side routing.

### Actual Behavior

It reloads every page from the server.

**Comment by Ehesp** [comment]
Do you have the Scripts component in your root?

**Comment by abereghici** [comment]
Yes, I have. I upgraded from 1.0.6 version and I encountered this problem. Also the `intent=prefetch` for the links reloads the page entirely.

**Comment by abereghici** [comment]
I created an abstraction for remix themes (https://github.com/abereghici/remix-themes) and in the theme provider I used `useFetcher` which caused the page to be reloaded each time I clicked on a link. I replaced `useFetcher` with a simple `fetch` and now is working as expected. What might be wrong here ? https://github.com/abereghici/remix-themes/blob/main/src/theme-provider.tsx#L60

**Comment by geocine** [comment]
I suggest you should update the original issue you raised and mention `useFetcher` specifically
